dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -52,25 +52,46 @@
   private static final RepositorySystem system = RepositoryUtility.newRepositorySystem();
   
   static {
-    // os.detected.classifier system property used to select Netty deps
-    String OS = System.getProperty("os.name", "generic").toLowerCase(Locale.ENGLISH);
-    if ((OS.indexOf("mac") >= 0) || (OS.indexOf("darwin") >= 0)) {
-      System.setProperty("os.detected.classifier", "osx-x86_64");
-    } else if (OS.indexOf("win") >= 0) {
-      System.setProperty("os.detected.classifier", "windows-x86_64");
-    } else if (OS.indexOf("nux") >= 0) {
-      System.setProperty("os.detected.classifier", "linux-x86_64");
-    } else {
-      // Since we only load the dependency graph, not actually use the 
-      // dependency, it doesn't matter a great deal which one we pick.
-      System.setProperty("os.detected.classifier", "linux-x86_64");      
-    }
+    setDetectedOsSystemProperties();
   }
 
   // caching cuts time by about a factor of 4.
   private static final Map<String, DependencyNode> cacheWithProvidedScope = new HashMap<>();
   private static final Map<String, DependencyNode> cacheWithoutProvidedScope = new HashMap<>();
 
+  private static void setDetectedOsSystemProperties() {
+    // System properties to select Netty dependencies through os-maven-plugin
+    // Definition of the properties: https://github.com/trustin/os-maven-plugin
+    String nonAlphaNumericPattern = "[^a-z0-9]+";
```

Comment:
Make this a constant? And maybe use `Pattern.compile()`?

Reply:
Now it's constant `private static final Pattern NON_ALPHA_NUMERIC_PATTERN = Pattern.compile("[^a-z0-9]+")`.